### DISCUSSION TITLE
Browsing Wifi is not working

### DISCUSSION THREAD

**Discussion by bnilsson11** [rern/rAudio#364] [Fixed Issues]
Browsing for wifi has not worked for a LONG time, if ever for rAudio.
Is this function dead and buried?

**Comment by R101G** [comment]
I must add that wifi generally seems to have become very fragile. My setup does not now survive a reboot.

**Comment by G0MJW** [comment]
Same here - the latest updates seem to have made reliability much worse than it was a year ago.

**Reply by G0MJW** [reply]
I should clarify, it keeps turning on the damned access point, despite this being disabled it gets re-enabled. Then it can't access Wifi. But worse, if there is a power outage - frequent here in the 3rd world - this access point gets detected as an internet link by my WiFi extenders because it comes uo advertising it's presence before the main router on power restore. Took me a while to discover what was happening. Perhaps there  are some files I can delete to prevent it becoming an access point, EVER.

**Reply by rern** [reply]
Image files just have been updated to the latest kernel and drivers. Wi-Fi connection should be better. My Zero, 3 and 4 all reconnected on reboot successfully 6 times in a row.

**Reply by R101G** [reply]
Thanks rern, downloading now. What is the future for the 64bit image?

**Reply by G0MJW** [reply]
What happens if you have no Wifi when re-booting? Say for 5-10 minutes? This is what happens after the power cuts.

**Reply by G0MJW** [reply]
OK - downloaded and installed the new image. How can I permanently stop it updating anything?

**Reply by R101G** [reply]
Going backwards here. I installed the driver for my wifi dongle to the new 32bit image, and set up everything as usual, but had no wifi at initial boot. I reverted to my 20210502 64bit image, which always works, but I now find that too fails on reboot.

So I too need to know if something is being automatically updated, and how to stop it if it is. I am having to keep several SD cards handy in case I have to power off.

**Reply by R101G** [reply]
OK. It looks as if rAudio will no longer connect to a hyphenated ssid after the initial boot.

**Reply by G0MJW** [reply]
Well I downloaded the new version yesterday, got it all working. Today the power went off so it could not connect to the router. It's no longer present on the network. 

Why is it not working? Because DESPITE ACCESS POINT BEING EXPLICITY DISABLED, it has STARTED UP IN ACCESS POINT MODE. Capitals to explain where to look for the bug. It does this when it can't access the network - not because the network is slow, it is because the network starts up more slowly than the PI.  

Now I have got it back, by power cycling just the PI, I can see access point is disabled. So why did it open an access point? It really should not do that when it is configured not to do so. Somewhere there is a process that is not looking at the configuration but instead doing it's own thing, perhaps getting board while waiting. That's what's probably behind all these WiFi reliability issues.

**Reply by R101G** [reply]
GoMJW

Are you sure it's not the same problem I have? If there are any characters in the SSID name which systemd does not like, it will only connect once. I think it's something to do with the way the netctl profile is associated with wlan by systemd has changed.

If I wait long enough with no wifi connection and no LAN, the AP activates. It's the most useful thing it can do :)

**Comment by R101G** [comment]


**Reply by G0MJW** [reply]
I think it expects WiFi to be there on reboot, often it is but when recovering from a power cut it isn't. Then it seems to turn on the access point and gets stuck and all you can do is pack it up, take it to a place where there is a wired network and connect to it that way, fiddle with files via SSH to fix it. 

Unfortunately the only way to update in the hope of a fix now seems to be to download the latest image and rebuild the SD card. That means effectively you can't keep your music library on the SD card.  Also after a few weeks it all breaks again automatically. I quite like Raudio but I am looking for a package that just works, that does not keep breaking itself with updates.

**Reply by ceejayemm** [reply]
How about separating your music library from the OS by putting the music on a separate USB drive and keeping a small SD card purely for the OS. Then the SD card can be refreshed with a new image whenever needed. If you create a backup before reflashing the SD card even the pain of rebuilding your library can be avoided too. This might also help with getting the wifi settings restored after a power cut ?

Chris

**Reply by G0MJW** [reply]
Obviously! That's why I said you can't keep your library on the SD card. There are two problems here. Firstly, that makes a less neat solution and with the PI Zero there is no spare USB port to plug a USB stick into. The 128, 256GB SD cards are relatively cheap. It doesn't help with the pain of power power cut fixing either. You basically have to connect an ethernet cable. So then you end up having to put an ethernet cable to the HiFi as in the UK power cuts are now a weekly occurrence. The living room ends up looking like an IT office.

**Reply by ceejayemm** [reply]
I have long kept a preconfigured wifi file (called 'wifi' with no extension) in a separate location (in my case on my NAS). When a problem with wifi access occurs the first thying I do is to copy this file to the /boot folder on my original SD card and let rAudio reboot from that. I find this overcomes most wifi connection problems I have encountered. The 'wifi' file contains:

Interface=wlan0
Connection=wireless
IP=dhcp
ESSID="XXXXXX"        <-- Replace with the SSID of your wifi network
Key=*******                 <-- Replace with the password key of your wifi network
Security=wpa

I too use a Pi Zero W and wanted a neat profile so I use:

Pimoroni Audio DAC SHIM (Line-Out)
https://shop.pimoroni.com/products/audio-dac-shim-line-out
(Supported by rAudio-1 as Pimoroni Audio DAC SHIM)

This provides a decent audio output and frees up a micro-usb port, I then use a power supply with a right angled micro-usb plug and:

Rankie R-1170 Micro USB (Male) to USB 2.0 (Female) Adapter, On-The-Go (OTG) Convertor Cable, 3-Pack, Black
https://amazon.co.uk/Rankie-R-1170-Female-Adapter-Convertor-Black/dp/B00YOX4JU6/

SanDisk Cruzer Fit 16GB USB 2.0 Flash Drive , Black
https://amazon.co.uk/SanDisk-Cruzer-16GB-Flash-Drive/dp/B07MDXBT87/

to provide the library data storage component. I also have a solid (ie no training wires) usb/micro-usb adaptor which I have had for ages and can't now think where I got it from hence the right angled power adapter plug.

Just out of interest which general part of the UK do you live in to be suffering such regular power outages ?  Do you seek compensation from your power supplier for such regular events ?  I too live in the UK and can't say that this is a regular occurrence for me, nor do I hear of it from friends in other areas across the whole of the UK.

Hope some of this helps.

Chris

**Comment by G0MJW** [comment]
No, there are only abc123 type characters in the SSID

**Reply by G0MJW** [reply]
It has happened again after todays power outage. At the moment, it isn't usable. Can anyone suggest an alternative? Access point disabled yet it is in access point mode.

**Reply by R101G** [reply]
I think switching to AP mode after a time with no connection was what always happened. The AP setting is for when you want permanent AP mode. I am pretty sure that a reboot used to restore everything to normal. From what you say, it looks as if the AP mode setting now sticks once it gets activated by the timeout. That would match my experience, although the reason for no connection was different.

Just to be clear, can you reboot the Pi normally, when there is no power cut, and the router is operating?

**Comment by G0MJW** [comment]
I can but it will do the same on the next power outage. How can we stop it ever starting up as an access point? Disabling access point out to do just that. Perhaps there is some legacy code?

**Reply by R101G** [reply]
I think this is one for @rern. Even if you figure out how to prevent the AP from starting, any changes you make will be overwritten by the next update. If you can make sure the Pi is unplugged while the power is off, and not powered up until the router is operating again, that might be a workaround in the meantime.

**Reply by DSYeow** [reply]
Thanks for the tip that the SSID can only contain alphanumeric characters.  I have rAudio 1 • 202106091 installed, and it refuses to connect to the Wifi AP due to the SSID containing a "-".  I tested with a spare AP and configured it to only alphanumeric and it works !  No issues rebooting and reconnecting to the spare AP.

Not sure if its for rern to figure out or Arch Linux but I sure hope there would be a solution - way too much hassle to change all the APs and reconfigure the wireless devices.  Thanks again.

**Reply by R101G** [reply]
DSYeow

From what I have read, this seems to be a problem unique to Netctl. I think it appeared when the command _systemctl enable netctl-auto@wlan0_ stopped working after an update. Before then, there was no problem.

**Reply by rern** [reply]
Just tested. SSID with `-` was connected via `/boot/wifi` and Networks page. Reboot reconnect OK as well.

`netctl-auto` - [ArchWiki](https://wiki.archlinux.org/title/netctl#Enabling_a_profile) ... profiles will be started/stopped automatically as you move from the range of one network into the range of another network (roaming).

Try `wifi-menu` command to connect the SSID with `-`.

**Reply by DSYeow** [reply]
Hi All,

Now this is weird.  I tried again after reading both R101 and rern's update.  I have setup a LAN connection as well so that I have a fall back connection in case the Wifi does not work.

When I powered up, the connection was on LAN.  I made sure my LAN setup was on DHCP.  Then went to Network and my saved settings of the SSID that had the "-" was still there.  Did a scan for Wifi networks - the SSID could not be found.  Granted the signal strength is marginal but it worked with the earlier RuneAudio+Re5 (did not try e6).  Since I have another Wifi AP which had stronger signal, I selected it.  This AP also had a "-" in the SSID.  And it worked !

Then I did a scan again and now the weaker signal AP appeared too and it worked as well.  Now that is puzzling.

Just note that before this,  I even tried setting it to FIXED IP and DHCP - it just refuses to connect.  Now, it works.

Hope it stays this way.  Thanks for all the responses and will update if I find something that I can identify as the cause.

**Reply by R101G** [reply]
Thanks for looking at this, rern. I found that the SSID connected using the boot/wifi file, but failed on a reboot. I have a hidden SSID, so if DSYeow does too, that might be the common factor. 

EDIT: wifi-menu only finds the connected 2.4GHz SSID. It does not show the 5GHz SSID on the same router, which has similar signal strength and no hyphen.

**Reply by rern** [reply]
Tested:
- Hidden SSID with `-`
    - Connected
    - Reboot and connected OK 5 times in a row.
- 5GHz SSID with `-` (not hidden)
    - Listed in Networks scan page
    - Listed in `wifi-menu` command
    - Connected

Try:
    - `pacman -Syu`
    - Reboot router

**Reply by R101G** [reply]
It should have been obvious to me that a hidden SSID would not show in wifi-menu - sorry.

**Reply by DSYeow** [reply]
Hi rern and R101-Github,

Sorry for the delay in replying. Had to do some checking on the network side of things, with my limited resources.
Used pretty simple stuff like inSSIDer, arp and ping to see what is going on.

- Just to confirm, No, the SSID is not hidden.  
- The "-" in the SSID name is not the issue.

When the Wifi signal is marginal, here is what I can see when I select that marginal Wifi connection :
- "arp -a" shows the ip address for the device
- ping device-ip-address has no response
- inSSIDer shows the signal while marginal it remains steady
- repeated cycles of this shows the same for a few minutes
- then "arp -a" shows the ip address for the device is no longer there (as if a reset is done)
- shortly after, "arp -a" shows the ip address is back up
- not too long after that, rAudio webpage is displayed and the system is up

When I then go to Networks and select another connection with good signal, the switch is fast
irrespective of whether I use DHCP or Static IP.

From the little that I can do, it appears when the signal is marginal, establishing a connection takes
much-much longer giving the impression that it has hung or refused to connect.  The odd part is it
actually tries again after a while (probably after some timeout) and it can get connected.  Why not the first time ?

I thought I would share what I discovered and basically, for peace of mind - either use cable or make sure the Wifi signal
is strong and there is nothing obstructing the RPi, especially those that have large metal heat sinks for passive heat dissipation.

Thanks and regards.

**Comment by bnilsson11** [comment]
The latest update stopped the notorious wifi enable and access point start. Thanks for that, good job!
However, browsing for available wifi networks is still not possible.
Is there any plan for this?

**Reply by rern** [reply]
@bnilsson11,
Post the output of:
`networks-scanwlan.sh | jq .`

**Reply by bnilsson11** [reply]
Just to avoid typo's:

rAudioRP4:~  #  networks-scanwlan.sh | jq .
wlan0: ERROR while getting interface flags: No such device
wlan0     Interface doesn't support scanning.

[]
rAudioRP4:~  #  networks-scanwlan.sh | jq  
wlan0: ERROR while getting interface flags: No such device
wlan0     Interface doesn't support scanning.

[]
rAudioRP4:~  #  networks-scanwlan.sh     
wlan0: ERROR while getting interface flags: No such device
wlan0     Interface doesn't support scanning.

[]

**Reply by bnilsson11** [reply]
Sorry, wifi must be enabled first...try again:

```
#  networks-scanwlan.sh | jq .
parse error: Invalid escape at line 1, column 49
```

**Reply by bnilsson11** [reply]
Found it.
I had used a guest wifi network name in my router which contained "international" characters.
I changed it to an English variant and now I can see my SSID networks and connect.
Works, case closed.

However, if it CAN be fixed I guess it should be fixed.
It seems it should not be a fundamental problem to have international characters in the names in the wifi networks. It was not a problem for the rest of my network infrastructure.

Thanks.
